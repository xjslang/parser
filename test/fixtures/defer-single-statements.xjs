function testSingleStatements() {
  console.log('start')
  
  const db = {
    close() {
      console.log('db closed')
    }
  }
  
  const file = {
    close() {
      console.log('file closed')
    }
  }
  
  // defer con sentencia única (sin llaves)
  defer db.close()
  
  // defer con bloque (con llaves) para comparar
  defer {
    file.close()
  }
  
  // defer con expresión más compleja
  defer console.log('cleanup done')
  
  console.log('end')
}

testSingleStatements()
