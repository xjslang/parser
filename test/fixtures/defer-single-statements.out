start
end
cleanup done
file closed
db closed
